realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -289,6 +291,23 @@ private boolean categorizeClassElements() {
                     }
                 }
 
+                if (variableElement.getAnnotation(NotNullable.class) == null) {
+                    // The field has not the @NotNullable annotation. It's only valid for:
+                    // * String
+                    String elementTypeCanonicalName = variableElement.asType().toString();
```

Comment:
The proper way to do this comparison would be to get the TypeMirror object for String like this:
```
TypeMirror stringType = env.getElementUtils().getTypeElement("java.lang.String").asType();
```
and then using that for comparison instead of comparing strings everywhere